### PR TITLE
Fix support for builtin renderers

### LINKED ISSUE #128451: Notebook renderer does not support application/json, text/x-html & text/x-<languageid>
This doesn't seem to work as documented https://code.visualstudio.com/api/extension-guides/notebook#rich-output

> Report Issue' dialog can assist with this. -->
- VS Code Version: Insiders 
- OS Version: N/A

Steps to Reproduce:

1. Create a notebook serializer and generate output using:
```javascript
* vscode.NotebookCellOutputItem.json({ hello: 'world' })
* vscode.NotebookCellOutputItem.text('console.log(123)', 'text/x-javascript'),
```

The error message in the generated output in the Notebook is  `No renderer could be found for MIME type: application/json`

Screen Shot 2021-07-11 at 20 37 40

@tanhakabir  @rebornix /cc

### PR DESCRIPTION
**Pull request by DonJayamanne**
This PR fixes #128451

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/browser/contrib/codeRenderer/codeRenderer.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export class NotebookCodeRendererContribution extends Disposable {
 }
 
 const workbenchContributionsRegistry = Registry.as<IWorkbenchContributionsRegistry>(WorkbenchExtensions.Workbench);
-workbenchContributionsRegistry.registerWorkbenchContribution(NotebookCodeRendererContribution, LifecyclePhase.Eventually);
+workbenchContributionsRegistry.registerWorkbenchContribution(NotebookCodeRendererContribution, LifecyclePhase.Restored);
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/browser/contrib/codeRenderer/codeRenderer.ts:99]: We need the renderers registered as soon as editors are restored (they start rendering)

File: src/vs/workbench/contrib/notebook/browser/view/output/outputRenderer.ts [hunk 2]
```diff
@@ -16,28 +16,34 @@ export class OutputRenderer {
 	private readonly _richMimeTypeRenderers = new Map<string, IOutputTransformContribution>();
 
 	constructor(
-		notebookEditor: ICommonNotebookEditor,
-		instantiationService: IInstantiationService
+		private readonly notebookEditor: ICommonNotebookEditor,
+		private readonly instantiationService: IInstantiationService
 	) {
-		for (const desc of OutputRendererRegistry.getOutputTransformContributions()) {
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/browser/view/output/outputRenderer.ts:22]: Building the list of renderers in the ctor will not work, as that's too early in the life cycle & other built-in renderers have not yet been regsitered.
We need to lazy load them (giving time for them to get registered)

**Comment by rebornix** [comment]
@DonJayamanne thanks for the contribution. I wonder if this really fixes the underling problem. We can have an offline discussion on this.

File: src/vs/workbench/contrib/notebook/browser/view/output/outputRenderer.ts (hunk 2 above)
Comment (rebornix) [reply src/vs/workbench/contrib/notebook/browser/view/output/outputRenderer.ts:22]: Other than postponing when `OutputRendererRegistry.getOutputTransformContributions` is called (which would never be correct now I think about is, as code renderers can register any time). We may want to move to the logic into `OutputRendererRegistry`.

**Review by rebornix — CHANGES_REQUESTED** [review]
I think we should always dynamic register code renderers (as they depend on text language contributions) and make sure that they are registered at the right time.

File: src/vs/workbench/contrib/notebook/browser/contrib/codeRenderer/codeRenderer.ts (hunk 1 above)
Comment (tanhakabir) [reply src/vs/workbench/contrib/notebook/browser/contrib/codeRenderer/codeRenderer.ts:99]: ah so restored happens after eventually? lgtm

**Comment by DonJayamanne** [comment]
> should always dynamic register code renderers

Or it should be a lazy evaluation.

**Pull request merged**